content/blaze.md

```diff
@@ -0,0 +1,543 @@
+
+<h2 id="reusable-components">Creating reusable components in Blaze</h2>
+In <a href="ui-ux">the UI/UX article</a> we discussed the merits of creating reusable components that are modular and depend and interact with their environment in clear and minimal ways. 
+
+Although Blaze, as a simple template-based runtime, doesn't enforce a lot of these principles (as some other rendering frameworks, such as React, do) you can take steps to write your Blaze components in such a way that you can enjoy most of the same benefits. This section will attempt to outline some "best practice" for writing such Blaze components.
+
+Examples below will reference the `listShow` component from the Todos example app.
+
+<h3 id="validate-data-context">Validate data contexts</h3>
+In order to ensure your component is only used in the way you expect, you should validate the data context provided to it. The data context provides the inputs to the component and so should be checked.
+
+XXX: this does not yet work
+
+You can do this in the components `onCreated()` callback, like so:
+
+```js
+Template.listsShow.onCreated(function() {
+  this.autorun(() => {
+    check(Template.data(), new SimpleSchema({
+     list: {blackbox: true},
+     todosReady: {type: Boolean}
+   }));
+  });
+```
+
+By placing the check in an `autorun()` we ensure that even if the data contexts reactively changes, it always fits the expected schema.
+
+<h3 id="name-data-contexts">Name data contexts to template inclusions</h3>
+It's tempting to provide the data context of a sub-template as a "raw" object (like `{{> todosItem todo}}`), it's a better idea to explicitly give it a name (`{{> todosItem todo=todo}}`). There are two primary reasons for this:
+
+1. When using the data in the sub-template, it's a lot clearer what you are accessing `{{todo.title}}` is clearer than `title`.
+2. It's more flexible, in case in future you need to provide more arguments to the template.
+
+  For instance, in the case of the `todosItem` sub-template, we need to provide two extra arguments to control the editing state of the item, which would have been a hassle to add if the item was used with a raw `todo` argument.
+
+Additionally, for similar reasons of clarity, always explicitly provide a data context to an inclusion, rather than letting it passively "fall-through".
+
+<h3 id="use-each-in">Prefer `{{#each .. in}}`</h3>
+For similar reasons to the above, it's better to use `{{#each todo in todos}}` rather than the older `{{#each todos}}`. The second sets the data context of it's internals to a raw `todo`, and makes it difficult to access the containing template's other data context. 
+
+The only reason not to use `{{#each .. in}}` because it makes it difficult to access the `todo` symbol inside event handlers. Typically the solution to this is simply to use a sub-component to render the inside of the loop.
```

Comment:
I think I'm pretty inconsistent in this article. I've used component more or less everywhere else in the guide. It's hard to know what to use in this article, sometimes it seems like it should be component (like when talking about "reusability"), other times template (when talking about technicalities with Blaze).  

This is sort of in-between.